capsules/extra/src/bus.rs

```diff
@@ -29,14 +29,34 @@
 
 use core::cell::Cell;
 use kernel::debug;
-use kernel::hil::bus8080::{self, Bus8080};
+use kernel::hil::bus8080::{self, Bus8080, BusAddr8080};
 use kernel::hil::i2c::{Error, I2CClient, I2CDevice};
 use kernel::hil::spi::{ClockPhase, ClockPolarity, SpiMasterClient, SpiMasterDevice};
 use kernel::utilities::cells::OptionalCell;
 use kernel::ErrorCode;
 
-/// Bus width used for address width and data width
-pub enum BusWidth {
+// Buses are typically serial (like I2C or SPI), transmitting data byte by byte
+// without considering endianness. The receiving device, in our case, the
+// screen, determines the endianness, as it interprets the data.
+// In Tock, the lower-level screen driver sets the address endianness.
+// For most buses, except the parallel 8080 bus, endianness remains transparent.
+
+// The bytes are stored in the proper endianness, ready to be sent on a serial bus.
+// Even if the bytes are in the proper order, we still need to know the endianness due
+// to buses, like the 8080, that are parallel as they might need to reconvert addresses
+// (the bus is wired in a way and the addresses are logically encoded in the other way).
+
+/// The `DataWidth` enum and related `BusAddr` structs represent both the width
+/// and endianness of data being transmitted over a bus.
+/// The byte indexing aligns with the endianness specified
+/// by the `DataWidth` enum:
+
+/// Index `0` always corresponds to the first byte that is transmitted.
+/// For big-endian, index `0` is the most significant byte.
+/// For little-endian, index `0` is the least significant byte.
+///
+/// The bytes are always stored in the proper endianess in the associated array.
```

Comment:
Same here, no longer using arrays.

Reply:
> I think you should copy (and adjust) that large document over DataWidth into the HIL and explain why the various widths & endianness selectors are required.

Since we no longer store the address in an array based on endianness, the comment explaining it has been revised to fit the new implementation and is now much shorter. Do you think we should still include it in the HIL, or is it fine as it is?